webapp/app/routes/app/freeze/create-freeze-proposal-modal.tsx

```diff
@@ -0,0 +1,111 @@
+import type { FreezeProposalsType } from "@/.server/db/schema";
+import { DateTimePicker } from "@/components/date-time-picker";
+import {
+  AlertDialog,
+  AlertDialogContent,
+  AlertDialogHeader,
+  AlertDialogTitle,
+} from "@/components/ui/alert-dialog";
+import { Button } from "@/components/ui/button";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import type { action } from "@/routes/app/freeze/_route";
+import { Cross2Icon, Share2Icon } from "@radix-ui/react-icons";
+import { Form, useActionData, useNavigation, useRevalidator } from "@remix-run/react";
+import { add } from "date-fns";
+import { useState } from "react";
+
+export function CreateFreezeProposalModal({
+  type,
+  onClose,
+}: {
+  type: FreezeProposalsType | null;
+  onClose: () => void;
+}) {
+  const actionResult = useActionData<typeof action>();
+  const [date, setDate] = useState<Date>(add(new Date(), { days: 7 }));
+  const navigation = useNavigation();
+  const revalidator = useRevalidator();
+
+  let title: string;
+  switch (type) {
+    case "SOFT_FREEZE":
+      title = "Create Soft Freeze Proposal";
+      break;
+    case "HARD_FREEZE":
+      title = "Create Hard Freeze Proposal";
+      break;
+    case "SET_SOFT_FREEZE_THRESHOLD":
+      title = "Create Set Soft Freeze Threshold Proposal";
+      break;
+    case "UNFREEZE":
+      title = "Create Unfreeze Proposal";
+      break;
+    default:
```

Comment:
Is this possible or it's an error?

Reply:
Do you mean the default option? Default should never be triggered.